### ISSUE TITLE
model. predict on loaded model: CatBoostError: json_helper.h:112: Error: wrong json type

### ISSUE THREAD

**Issue by vale9003** [catboost/catboost#1321]
Problem: Error: wrong json type, when using predict method on a loaded model
catboost version: 0.23
Operating System: Saved on Linux loaded on Windows
CPU: core i7

I have saved a catboost model as I always do on azure databricks using dill, I have also tried with catboost save model, I have downloaded the compiled model from databricks to my local pc, and when I have to use the model to predict, that's when the error is thrown. Error: wrong json type

I need help ASAP.  I don't know if this is a catboost issue or a problem with databricks, but it didn't happen before. I didn't change anything to the way I was working.

Best Regards,
Valentina.

**Comment by coolcodecding** [comment]
I am facing the same problem using 
    filename = './saved_models/catboost.sav'
    pickle.dump(model, open(filename, 'wb'))

when I load the model and try to use
model.predict(X)
I get:

<ipython-input-5-ef4b60251f77> in evaluate_model(X, y, model)
      1 def evaluate_model(X, y, model):
----> 2     y_pred = model.predict(X)
      3 
      4     clf_report = classification_report(y, y_pred, output_dict = True)
      5     report = pd.DataFrame(clf_report)

~/anaconda3/envs/dipla/lib/python3.8/site-packages/catboost/core.py in predict(self, data, prediction_type, ntree_start, ntree_end, thread_count, verbose)
   4011                   with log probability for every class for each object.
   4012         """
-> 4013         return self._predict(data, prediction_type, ntree_start, ntree_end, thread_count, verbose, 'predict')
   4014 
   4015     def predict_proba(self, data, ntree_start=0, ntree_end=0, thread_count=-1, verbose=None):

~/anaconda3/envs/dipla/lib/python3.8/site-packages/catboost/core.py in _predict(self, data, prediction_type, ntree_start, ntree_end, thread_count, verbose, parent_method_name)
   1876         self._validate_prediction_type(prediction_type)
   1877 
-> 1878         predictions = self._base_predict(data, prediction_type, ntree_start, ntree_end, thread_count, verbose)
   1879         return predictions[0] if data_is_single_object else predictions
   1880 

~/anaconda3/envs/dipla/lib/python3.8/site-packages/catboost/core.py in _base_predict(self, pool, prediction_type, ntree_start, ntree_end, thread_count, verbose)
   1273 
   1274     def _base_predict(self, pool, prediction_type, ntree_start, ntree_end, thread_count, verbose):
-> 1275         return self._object._base_predict(pool, prediction_type, ntree_start, ntree_end, thread_count, verbose)
   1276 
   1277     def _staged_predict_iterator(self, pool, prediction_type, ntree_start, ntree_end, eval_period, thread_count, verbose):

_catboost.pyx in _catboost._CatBoost._base_predict()

_catboost.pyx in _catboost._CatBoost._base_predict()

CatBoostError: catboost/private/libs/options/json_helper.h:112: Error: wrong json type

**Comment by kizill** [comment]
Hi all!
Sorry for no reaction in this issue, i've forgottent to answer here but already fixed that in 0.23.2.There was an unintended cross version model serialization incompatibility bug. Please update to 0.23.2 where we revert backwards compatibility and feel free to reopen this issue if problem persists.

**Comment by pushpankar** [comment]
Hello,
I am using 0.23.2 but still getting the issue:
 ```
Traceback (most recent call last):
  File "/mnt/vol1/anaconda3/lib/python3.7/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/mnt/vol1/anaconda3/lib/python3.7/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/mnt/vol1/anaconda3/lib/python3.7/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/mnt/vol1/anaconda3/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/mnt/vol1/anaconda3/lib/python3.7/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/mnt/vol1/anaconda3/lib/python3.7/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/pushpankar.kumar/moderation_api/api.py", line 69, in is_safe
    result = predict(content, result)
  File "/home/pushpankar.kumar/moderation_api/model/tree_model.py", line 16, in predict
    prediction = model.predict(features[2:])[0]
  File "/mnt/vol1/anaconda3/lib/python3.7/site-packages/catboost/core.py", line 3837, in predict
    return self._predict(data, prediction_type, ntree_start, ntree_end, thread_count, verbose, 'predict')
  File "/mnt/vol1/anaconda3/lib/python3.7/site-packages/catboost/core.py", line 1849, in _predict
    predictions = self._base_predict(data, prediction_type, ntree_start, ntree_end, thread_count, verbose)
  File "/mnt/vol1/anaconda3/lib/python3.7/site-packages/catboost/core.py", line 1268, in _base_predict
    return self._object._base_predict(pool, prediction_type, ntree_start, ntree_end, thread_count, verbose)
  File "_catboost.pyx", line 3855, in _catboost._CatBoost._base_predict
  File "_catboost.pyx", line 3860, in _catboost._CatBoost._base_predict
_catboost.CatBoostError: catboost/private/libs/options/json_helper.h:112: Error: wrong json type 

```
Here is my `conda list` :
``` c-ares                    1.16.1               h7b6447c_0
ca-certificates           2020.7.22                     0
cachetools                4.1.1                    pypi_0    pypi
catboost                  0.23.2                   pypi_0    pypi
certifi                   2020.6.20                py37_0
cffi                      1.14.3           py37he30daa8_0
chardet                   3.0.4                 py37_1003
click                     7.1.2                      py_0
cloudpickle               1.6.0                    pypi_0    pypi
colorthief                0.2.1                    pypi_0    pypi
conda                     4.8.5                    py37_0
conda-package-handling    1.6.1            py37h7b6447c_0
 ```
